example/newreposecret/main.go

```diff
@@ -0,0 +1,197 @@
+package main
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"flag"
+	"fmt"
+	sodium "github.com/GoKillers/libsodium-go/cryptobox"
+	"github.com/google/go-github/v32/github"
+	"golang.org/x/oauth2"
+	"log"
+	"os"
+)
+
+var (
+	repo  = flag.String("repo", "", "The repo that the secret should be added to, ex. go-github")
+	owner = flag.String("owner", "", "The owner of there repo this should be added to, ex. google")
+)
+
+/*
+main creates a new secret in github for a given owner/repo based on a secretName passed in as an arg
+and a github token, and secret value provided via the environment
+
+usage:
+	export GITHUB_AUTH_TOKEN=<auth token from github that has secret create rights>
+	export SECRET_VARIABLE=<secret value of the secret variable>
+	go run main.go -owner <owner name> -repo <repository name> SECRET_VARIABLE
+
+ex:
+	export GITHUB_AUTH_TOKEN=0000000000000000
+	export SECRET_VARIABLE="my-secret"
+	go run main.go -owner google -repo go-github SECRET_VARIABLE
+
+Once it runs go to the github repository > settings > left side options bar > Secrets
+And you should see the new secret appear there.
+*/
+func main() {
+	flag.Parse()
+
+	token := os.Getenv("GITHUB_AUTH_TOKEN")
+	if token == "" {
+		log.Fatal(`Unauthorized: No token present in environment variables. 
```

Comment:
Let's please make each `log.Fatal` message just one line long if possible. Hopefully the documentation up at the top or the comments will explain all the details

Reply:
updated!